tests/decrosstalk/test_io_utils.py

```diff
@@ -0,0 +1,46 @@
+import os
+import numpy as np
+import tempfile
+import h5py
+import ophys_etl.decrosstalk.io_utils as io_utils
+
+from .utils import teardown_function  # noqa F401
+from .utils import get_tmp_dir
+
+
+def test_write_to_h5():
+
+    test_write_to_h5._temp_files = []
+
+    data = {}
+    data['first_dir'] = 2
+    data['second_dir'] = {}
+    data['second_dir']['sub1'] = False
+    data['second_dir']['sub2'] = np.array([9, 4, 5, 1])
+    data['second_dir']['sub3'] = {}
+    data['second_dir']['sub3']['a'] = 4.5
+    data['second_dir']['sub3']['b'] = np.array([1.2, 4.3])
+    data['second_dir']['sub3']['c'] = True
+
+    tmp_dir = get_tmp_dir()
```

Comment:
i don't think this should be used. You are assuming you have write access to the `tmp` dir in the repo. pytest has the `tmp_path` fixture for such things:
https://docs.pytest.org/en/stable/tmpdir.html

Reply:
also, should eliminate need for `teardown_function`

Reply:
How bad is the assumption that whoever is running the tests can write into `tests/decrosstalk/tmp/`?

I'm playing around with
```
import pytest

def test_junk(tmpdir):
    tmpdir.mkdir()
    pass
```

It is producing a weird "File exists" error, both on my Macbook and on synapse. pytest claims that the bug is fixed

https://github.com/pytest-dev/pytest/issues/3451

Obviously not, though.

Reply:
bad enough that I have had this come up in the past.
`tmp_path` rather than `tmpdir` I think will take care of all of this for you.
Example:
https://github.com/AllenInstitute/ophys_etl_pipelines/blob/94d8610013dae5f1a4e67249b5654bd64c13308b/tests/transforms/test_suite2p_wrapper.py#L28-L42

Reply:
I can't recall the exact failure mode, but, I think it had something to do with certain priveleges while building a conda env inside of a dockerfile, and then wanting to run the tests inside the built docker image.

Reply:
Thanks for the help. I obviously don't have a lot of experience with pytest.

I've implemented this change.